src/Components/Tasks/Tasks.tsx

```diff
@@ -1,129 +1,216 @@
-      .then((response: any) => {
-        console.log(response.data.data);
-        setTasks(response?.data?.data);
+  {/*Update Task */ }
+  const editTaskShow =(id)=>{
+    
+    navigate(`/dashboard/edit-task/${id}`)
 
-      })
-      .catch((error:any) => {
-        console.log("error",error.response.data.data.message);
+  }
 
-      });
-  };
+  const getEmployeeTasks =()=>{
+    axios.get(`${baseUrl}/Task`,{headers:reqHeaders}).then((response)=>{
+      console.log(response);
+      
+  
+      const todosTask=response?.data?.data.filter((task:any)=>task.status=='ToDo')
+      const inProgressTask=response?.data?.data.filter((task:any)=>task.status=='InProgress')
+      const doneTask=response?.data?.data.filter((task:any)=>task.status=='Done')
+      setTodos(todosTask)
+      setInProgress(inProgressTask)
+      setDone(doneTask)
+  
+      setTasks(response.data.data)
+   
+    })
+   }
+  
 
   useEffect(() => {
+    if(role==='Manager')
     getTasksList();
+  else
+  getEmployeeTasks()
   }, []);
 
+
+  const statuses =['ToDo','InProgress','Done']
   return (
     <>
-         <Modal show={modalState==='delete-modal'&& "update-modal"} onHide={handleClose}>
+  
+      <Modal show={modalState === 'delete-modal'} onHide={handleClose}>
 
-         <Modal.Body>
-       <div className="delete-container">
-          <div className="icons text-end">
-        <i onClick={handleClose} className="fa-regular fa-circle-xmark text-danger "></i>
-          </div>
-          <div className="text-center">
+        <Modal.Body>
+          <div className="delete-container">
+            <div className="icons text-end">
+              <i onClick={handleClose} className="fa-regular fa-circle-xmark text-danger "></i>
+            </div>
             <div className="text-center">
-            <img className=' '  src={noData} alt="msg-NoData" />
+              <div className="text-center">
+                <img  src={Datano} alt="msg-NoData" />
+              </div>
+              <h5 className='py-3'> Are you sure to Delete this item ? </h5>
             </div>
-            <h5 className='py-3'> Are you sure to Delete this item ? </h5>
-          </div>
 
-          <div className="delete-btn text-end">
-            <button  className='text-white bg-danger btn btn-outline-danger   border-danger p-3  '>Delete This Item </button>
+            <div className="delete-btn text-end">
+              <button onClick={deleteTask} className='text-white bg-danger btn btn-outline-danger   border-danger p-3  '>Delete This Item </button>
+            </div>
           </div>
+        </Modal.Body>
+      </Modal>
+      <div className="container bg-white rounded-3 px-3">
+        <div className="col-md-12 trans-head">
+          <div className="top d-flex justify-content-between p-3">
+            <h3>Tasks</h3>
+            {role == "Manager" ? <button onClick={navToAddTask} className='btn bg-warning text-white'> <i className='fa-plus'></i> Add New Task </button>
```

Comment:
instead of ternary operator, you can change this to the following:
```
 {role == "Manager"  && (
<button onClick={navToAddTask} className='btn bg-warning text-white'> 
<i className='fa-plus'></i> Add New Task 
</button>)}
```
``